src/main/java/org/reactome/release/downloaddirectory/MapOldStableIds.java

```diff
@@ -79,50 +83,107 @@ public static void execute(String username, String password, String host, String
+	 * their associated instance ids that have ever existed in Reactome) to build a map of instance IDs to StableIdentifiers.
+	 * @param stableIdResults ResultSet - Data result of query to stable_identifiers database for stable identifiers and associated instance IDs.
+	 * @return Map<String, List<String>> - Mapping of db IDs to Stable Identifiers.
+	 * @throws SQLException - Thrown if there are issues accessing the ResultSet object.
+	 */
+	private static Map<String, List<String>> getDbIdToStableIds(ResultSet stableIdResults) throws SQLException {
 		Map<String, List<String>> dbIdToStableIds = new HashMap<>();
 
 		// Iterate through returned results of DB IDs and stable IDs
-		while (resultSet.next()) {
-			String stableId = resultSet.getString(1);
-			String dbId = resultSet.getString(2);
+		while (stableIdResults.next()) {
+			String stableId = stableIdResults.getString(1);
+			String dbId = stableIdResults.getString(2);
 
-			if (dbIdToStableIds.get(dbId) != null) {
-				dbIdToStableIds.get(dbId).add(stableId);
-			} else {
-				List<String> stableIds = new ArrayList<>(Collections.singletonList(stableId));
-				dbIdToStableIds.put(dbId, stableIds);
-			}
+			dbIdToStableIds.computeIfAbsent(dbId, k -> new ArrayList<>()).add(stableId);
 		}
-
 		return dbIdToStableIds;
 	}
+
+	/**
+	 * With the old stable identifier mappings completed, write the results to the 'reactome_stable_ids.txt' file.
+	 * @param releaseNumber String - Current release number, used for storing files in release-specific location.
+	 * @param stableIdsToOldIdsMappings List<List<Object>> - List of current stable identifier mappings to older mappings.
+	 * The interior List<Object> is of the form [String, List<String>].
+	 * @param currentStableIdentifiers Set<String>, all StableIdentifiers in current release database.
+	 * @throws IOException - Thrown if there are issues with creating or moving mapping file.
+	 */
+	private static void writeMappingsToFile(String releaseNumber, List<List<Object>> stableIdsToOldIdsMappings, Set<String> currentStableIdentifiers) throws IOException {
+		String header = "# Reactome stable IDs for release " + releaseNumber + "\n" + "Stable_ID\told_identifier(s)\n";
+		Files.write(Paths.get(filename), header.getBytes(), StandardOpenOption.APPEND);
+		for (List<Object> stableIdsArray : stableIdsToOldIdsMappings)
+		{
+			String primaryId = (String) stableIdsArray.get(0);
+			@SuppressWarnings("unchecked")
+			List<String> secondaryIds = (ArrayList<String>) stableIdsArray.get(1);
+			if (currentStableIdentifierWithMapping(currentStableIdentifiers, primaryId, secondaryIds)) {
+				String line = primaryId + "\t" + String.join(",", secondaryIds) + "\n";
+				Files.write(Paths.get(filename), line.getBytes(), StandardOpenOption.APPEND);
+			}
+		}
+		String outpathName = releaseNumber + "/" + filename;
+		Files.move(Paths.get(filename), Paths.get(outpathName), StandardCopyOption.REPLACE_EXISTING);
```

Comment:
This could be:
`Files.move(Paths.get(filename), Paths.get(releaseNumber, filename), StandardCopyOption.REPLACE_EXISTING);`

Reply:
Also, out of curiosity, could the file be written directly at the `releaseNumber/filename` path instead of moving it after writing at the `filename` path?

Reply:
Yea I did that, originally I had the DD files being made elsewhere and then moved to the `releaseNumber` directory, but in this case there isn't a reason not to just populate it there